### PR TITLE
[Serializer] ObjectNormalizer: don't serialize static methods and props

### PR DESCRIPTION
**Pull request by dunglas**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #16485 |
| License | MIT |
| Doc PR | n/a |

### CODE REVIEW

**Comment by dunglas** [comment]
Maybe should also open another PR for the PropertyAccess Component to prevent it from trying to access static methods/properties?

File: src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php [hunk 1]
```diff
@@ -66,8 +66,9 @@ public function normalize($object, $format = null, array $context = array())
 
             // methods
             $reflClass = new \ReflectionClass($object);
-            foreach ($reflClass->getMethods(\ReflectionMethod::IS_PUBLIC) as $reflMethod) {
+            foreach ($reflClass->getMethods(\ReflectionMethod::IS_PUBLIC ) as $reflMethod) {
```
Comment (Tobion) [inline src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php]: wrong cs change

**Comment by dunglas** [comment]
CS fixed. Travis failure is not related.

**Comment by dunglas** [comment]
ping @symfony/deciders

**Comment by roundge** [comment]
yeah thanks @dunglas to take seriously my problem!

**Comment by stof** [comment]
makes sense to me

**Comment by Tobion** [comment]
:+1: 

Status: Reviewed

**Comment by xabbuh** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @dunglas.

**Pull request closed without merging**